### ISSUE TITLE
Injecting JavaToolchainService into plugin fails

### ISSUE THREAD

**Issue by Vampire** [gradle/gradle#15819]
https://docs.gradle.org/current/userguide/toolchains.html states
> By injecting the JavaToolchainService in the plugin or task, it is also possible to wire a convention in those properties by obtaining the JavaToolchainSpec from the java extension on the project. 

So I tried that in 6.8, I have added to my plugin
```kt
@get:Inject
abstract val javaToolchains: JavaToolchainService
```
and then used it like
```kt
val myTask = project.tasks.register<MyTask>("myTask$foo") {
    barExecutable.convention(
        layout.file(
            javaToolchains
                .compilerFor {
                    languageVersion.set(JavaLanguageVersion.of(JavaVersion.current().majorVersion))
                }
                .map {
                    it.executablePath.asFile.resolveSibling("bar")
                }
        )
    )
}
```
But this does not work, it fails with `Caused by: org.gradle.internal.service.UnknownServiceException: No service of type JavaToolchainService available in ProjectScopeServices.` at `org.gradle.internal.service.DefaultServiceRegistry.get(DefaultServiceRegistry.java:291)`.

If I use `project.extensions.getByType<JavaToolchainService>()` instead of the inject, it works just fine.

**Comment by Tapchicoma** [comment]
FYI: according to my tests `JavaToolchainService` is injectable starting from Gradle 7

**Comment by bmuskalla** [comment]
This has been fixed in 245d9e7744596d27dee648fb8617eeb1fe6bc6dd for 7.0
